### PR TITLE
Basic setup

### PR DESCRIPTION
**Pull request by changchup**

### CODE REVIEW

File: API/__tests__/App.test.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        country: 'Canada'
+      }
+    }
+
+    await request(app)
+      .post("/")
+      .send(payload)
+      .expect(200)
+      .expect((response) => {
+        expect(response.body).toEqual({ result: '[]' });
```
Comment (antonkatz) [inline API/__tests__/App.test.ts:26]: Empty array is an unclear result.

File: API/__tests__/App.test.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    const payload = {}
+
+    await request(app)
+      .post("/")
+      .send(payload)
+      .expect(200)
+      .expect((response) => {
+        
+        console.log()
+        expect(JSON.parse(response.body.result).length ===8).toBeTruthy();
```
Comment (antonkatz) [inline API/__tests__/App.test.ts:41]: What's the error supposed to be?

File: API/__tests__/validate.test.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+        country: "test"
+      }
+    }
+    const errors = await validate(data)
+    expect(errors.length === 0).toBeTruthy()
+  });
+
+  it("should not validate", async () => {
+    const errors = await validate({})
+    expect(errors.length === 8).toBeTruthy()
```
Comment (antonkatz) [inline API/__tests__/validate.test.ts:24]: There are 8 errors?

File: API/src/validate.ts [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+  try {
+    await UserSchema.validate(data, { abortEarly: false });
+  } catch (err: any) {
+    for (const e of err.inner) {
+      errors.push({
+        name: e.path,
+        message: e.message
+      });
+    };
+    return errors;
```
Comment (antonkatz) [inline API/src/validate.ts:16]: Use a `finally` clause? This `return` is redundant.

File: UI/src/App.tsx [hunk 5]
```diff
@@ -1,24 +1,102 @@
...
+        ...prevState.address,
+        [e.target.id]: e.target.value,
+      },
+    }));
+  };
+
+  const onSubmit = async (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    console.log(formData);
+    const response = await fetch("http://localhost:3000",
```
Comment (antonkatz) [inline UI/src/App.tsx:45]: Not going to work outside dev env

File: UI/src/App.tsx [hunk 6]
```diff
@@ -1,24 +1,102 @@
...
-        >
-          Learn React
-        </a>
-      </header>
+    <div>
+      <h1>Form</h1>
+      <p>User Details</p>
+      <form onSubmit={onSubmit}>
+        <label htmlFor="name">Name</label>
+        <br />
```
Comment (antonkatz) [inline UI/src/App.tsx:65]: I would rely on css over `br` to break the form into lines

File: interfaces.ts [hunk 7]
```diff
@@ -0,0 +1,19 @@
+export interface FieldError {
+  name: string,
+  message: string
+}
+
+export interface Address {
```
Comment (antonkatz) [inline interfaces.ts:6]: Duplicates? `UI/src/interfaces.ts` has same definitions

File: API/__tests__/App.test.ts [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+      email: 'hamisha@hotmail.com',
+      phone_number: '7822343016',
+      address: {
+        house_number: '2059',
+        street_name: 'Beech Street',
+        city: 'Halifax',
+        state: 'Nova Scotia',
+        country: 'Canada'
+      }
+    }
```
[the remark this answers is not in the document]
Comment (changchup) [reply API/__tests__/App.test.ts:19]: noted

File: API/__tests__/validate.test.ts (hunk 3 above)
Comment (changchup) [reply API/__tests__/validate.test.ts:24]: this is testing to make sure the data structure is correctly validating a correct data structure and finding errors on an incorrect one, not testing the implementation of yup.

**Pull request merged**